text/3700-ptr-tag-helpers.md

```diff
@@ -0,0 +1,128 @@
+- Feature Name: `ptr_tag_helpers`
+- Start Date: 2024-09-26
+- RFC PR: [rust-lang/rfcs#3700](https://github.com/rust-lang/rfcs/pull/3700)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Add helper methods on primitive pointer types to facilitate getting and setting the tag of a pointer.
+Intended to work with programs that make use of architecture features such as AArch64
+Top-Byte Ignore (TBI), the primary use-case being writing tagging memory allocators.
+
+# Motivation
+[motivation]: #motivation
+
+Tagged pointers are pointers in which the unused top bits are set to contain some metadata - the tag.
+No 64-bit architecture today actually uses a 64-bit address space. Most operating systems only use
+the lower 48 bits, leaving higher bits unused. The remaining bits are for the most part used to
+distinguish userspace pointers (0x00) from kernelspace pointers (0xff).
+Certain architectures provide extensions, such as TBI on AArch64, that allow programs to make use of
+those unused bits to insert custom metadata into the pointer.
```

Comment:
To be clear, userspace can already do that without hardware support, trivially. By masking out those bots for each load. The hardware feature just makes this slightly more efficient.

I find the introduction to be a bit confusing due to this.